frontend/packages/console-app/src/components/nodes/NodesPage.tsx

```diff
@@ -40,60 +56,108 @@ const NodeTableHeader = () => {
-      </TableData>
-    </TableRow>
-  );
+const mapStateToProps = ({ UI }) => ({
+  metrics: UI.getIn(['metrics', 'node']),
+});
+
+type NodesRowMapFromStateProps = {
+  metrics: NodeMetrics;
 };
+
+const NodesTableRow = connect<NodesRowMapFromStateProps, null, NodesTableRowProps>(mapStateToProps)(
+  ({ obj: node, index, key, style, metrics }: NodesTableRowProps & NodesRowMapFromStateProps) => {
+    const nodeName = getName(node);
+    const nodeUID = getUID(node);
+    const freeMem: number = Number(_.get(metrics, ['freeMemory', nodeName]));
+    const totalMem: number = Number(_.get(metrics, ['totalMemory', nodeName]));
+    const usedMemory: string = humanizeBinaryBytes(totalMem - freeMem).string;
+    const totalMemory: string = humanizeBinaryBytes(totalMem).string;
+    const cores: number = Number(_.get(metrics, ['cpu', nodeName]));
+    const freeStrg: number = Number(_.get(metrics, ['freeStorage', nodeName]));
+    const totalStrg: number = Number(_.get(metrics, ['totalStorage', nodeName]));
+    const usedStorage: string = humanizeBinaryBytes(totalStrg - freeStrg).string;
+    const totalStorage: string = humanizeBinaryBytes(totalStrg).string;
+    return (
+      <TableRow id={nodeUID} index={index} trKey={key} style={style}>
+        <TableData className={tableColumnClasses[0]}>
+          <ResourceLink kind={referenceForModel(NodeModel)} name={nodeName} title={nodeUID} />
+        </TableData>
+        <TableData className={tableColumnClasses[1]}>
+          <NodeStatus node={node} />
+        </TableData>
+        <TableData className={tableColumnClasses[2]}>
+          <NodeRoles node={node} />
+        </TableData>
+        <TableData className={tableColumnClasses[3]}>{node.status?.capacity?.pods}</TableData>
+        <TableData className={tableColumnClasses[4]}>{`${usedMemory}/${totalMemory}`}</TableData>
+        <TableData className={tableColumnClasses[5]}>
+          {cores ? `${formatCores(cores)} cores` : '-'}
+        </TableData>
+        <TableData className={tableColumnClasses[6]}>{`${usedStorage}/${totalStorage}`}</TableData>
+        <TableData className={tableColumnClasses[7]}>
+          {fromNow(node.metadata?.creationTimestamp)}
+        </TableData>
+        <TableData className={tableColumnClasses[8]}>
+          <ResourceKebab
+            actions={menuActions}
+            kind={referenceForModel(NodeModel)}
+            resource={node}
+          />
+        </TableData>
+      </TableRow>
+    );
+  },
+);
 NodesTableRow.displayName = 'NodesTableRow';
+
 type NodesTableRowProps = {
   obj: NodeKind;
   index: number;
   key?: string;
   style: object;
 };
 
-const NodesTable: React.FC<NodesTableProps> = (props) => (
-  <Table {...props} aria-label="Nodes" Header={NodeTableHeader} Row={NodesTableRow} virtualize />
-);
+const NodesTable: React.FC<NodesTableProps> = (props) => {
+  const row = (rowProps: NodesTableRowProps) => <NodesTableRow {...rowProps} />;
```

Comment:
I suspect this is going to cause bad performance problems because the `row` is different on every render. Can this be pulled outside of the component?